misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -15,24 +15,283 @@
+     * @throws FormatterException if the file resolution fails
+     */
+    public static Map<String, Object> getFormattingConfigurations(Path root, String configurationFilePath)
+            throws FormatterException {
+        String content;
+        Optional<Path> path = convertConfigurationPath(configurationFilePath);
+        Optional<Path> absPath = path.isEmpty() || path.get().isAbsolute() || !root.isAbsolute() ? path :
+                Optional.of(root.resolve(configurationFilePath));
+        if (isLocalFile(absPath)) {
+            try {
+                content = Files.readString(absPath.get(), StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                throw new FormatterException("Failed to retrieve local formatting configuration file");
+            }
+        } else {
+            content = readRemoteFormatFile(root, configurationFilePath);
+        }
+
+        return parseConfigurationToml(TomlDocument.from(configurationFilePath, content));
+    }
+
+    private static Optional<Path> convertConfigurationPath(String path) {
+        try {
+            return Optional.of(Paths.get(path));
+        } catch (InvalidPathException ex) {
+            return Optional.empty();
+        }
+    }
+
+    private static boolean isLocalFile(Optional<Path> path) throws InvalidPathException {
+        return !path.isEmpty() && new File(path.get().toString()).exists();
+    }
+
+    private static String readRemoteFormatFile(Path root, String fileUrl) throws FormatterException {
+        Path cachePath = root.resolve(TARGET_DIR).resolve(FORMAT).resolve(DEFAULT_FORMAT_OPTION_FILE);
+        if (Files.exists(cachePath)) {
+            try {
+                return Files.readString(cachePath, StandardCharsets.UTF_8);
+            } catch (IOException e) {
+                throw new FormatterException("Failed to read cached formatting configuration file");
+            }
+        }
+
+        StringBuilder fileContent = new StringBuilder();
+        try {
+            URL url = new URL(fileUrl);
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            int responseCode = connection.getResponseCode();
+            if (responseCode != HttpURLConnection.HTTP_OK) {
+                connection.disconnect();
+                throw new FormatterException("Failed to retrieve remote file. HTTP response code: " + responseCode);
+            }
+            try (BufferedReader reader = new BufferedReader(
+                    new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    fileContent.append(line).append(NEWLINE_SYMBOL);
+                }
+            }
+            connection.disconnect();
+        } catch (IOException e) {
+            throw new FormatterException("Failed to retrieve formatting configuration file");
```

Comment:
When the file that cannot be found is related to what is specified in the Ballerina.toml file, can we mention that? 

We can probably do this separately, but may help in case the user can't figure out why we are looking for such a file.